aiohttp_client_cache/backends/filesystem.py

```diff
@@ -72,9 +72,13 @@ async def contains(self, key: str) -> bool:
         return isfile(self._join(key))
 
     async def read(self, key: str) -> ResponseOrKey:
-        with self._try_io():
-            async with aiofiles.open(self._join(key), 'rb') as f:
-                return self.deserialize(await f.read())
+        with self._try_io(False):
+            path = self._join(key)
+            if await aiofiles.os.path.exists(path):
```

Comment:
What exception type did you get without this check? Everything I've encountered so far inherits from `OSError`. I'd prefer to catch exceptions in `_try_io()` rather then checking first, if possible.

Other than that, this PR looks good. Thanks for the updates!

Reply:
I got an OSError as well, I suspect that something with the async with aiofile.open has unexpected side-effects leading to the problem. This snippet avoids that all together.

Reply:
Its maybe as in this a return value is expected, however the original snippet does not return something when the open fails with an exception. This might lead to weird side-effects with asyncio, I did not find a way to add a return None in the existing code when the block would return as my ide shows an unreachable code warning. My solution does that in a clean way.

Reply:
Alright, that works for me! Thanks.